### ISSUE TITLE
Used but not assigned not warning when in same line

### ISSUE THREAD

**Issue by rugk** [koalaman/shellcheck#1083]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
hu="x$hu"
number=$(( number+1 ))
```

#### Here's what shellcheck currently says:

nothing

The problem is, it does consider that the right part of the `=` is evaluated first and thus the variable is *first* used and later assigned to itself. Especially with arithmetic operations, this is highly problematic. 

#### Here's what I wanted or expected to see:

Warning: "Variable hu referenced before assigned."
Warning: "Variable number referenced before assigned."

**Comment by nihilus** [comment]
How is that problematic in arithmetic operations? It will implicity get the value 0.

**Comment by rugk** [comment]
If you think of multiplication (`df*s`) it will get worse (syntax errored in my small test) out dividing through zero it so…

**Comment by koalaman** [comment]
You're right, of course. This is something I really want to fix in the general case, but it requires some significant infrastructure improvements that I haven't gotten around to (specifically, building a useful control flow graph of the script).

**Comment by NateEag** [comment]
I spent a few minutes yesterday scratching my head until I realized I had made this error.

It surprised me that Shellcheck didn't catch it for me, as it does amazingly well at detecting my stupid mistakes overall.

@koalaman , when you say you want to fix this "in the general case", do you mean to imply there's a simpler, dumber fix for this that you don't want to use, because analogous problems will exist elsewhere?
